lib/mysql2/client.rb

```diff
@@ -83,8 +88,11 @@ def initialize(opts = {})
       port = port.to_i unless port.nil?
       database = database.to_s unless database.nil?
       socket = socket.to_s unless socket.nil?
+      conn_attrs = conn_attrs.each_with_object({}) do |(key, value), hash|
+        hash[key.to_sym] = value.to_s
```

Comment:
If this were `hash[key.to_s] = value.to_s` and avoided symbols, it looks like a lot of code in `opt_connect_addr_add_i` could be removed, and just StringValueToCStr used for the key?